docs/tools/brilift.md

```diff
@@ -0,0 +1,63 @@
+Cranelift Compiler
+==================
+
+Brilift is a compiler from Bril to native code using the [Cranelift][] code generator.
+It supports [core Bril][core] only.
+
+Brilift is an ahead-of-time compiler.
+It emits `.o` files and also provides a simple run-time library.
+By linking these together, you get a complete native executable.
+
+[cranelift]: https://github.com/bytecodealliance/wasmtime/tree/main/cranelift
+[core]: ../lang/core.md
+
+Build
+-----
+
+Brilift is a Rust project using the [bril-rs][] library.
+You can build it using [Cargo][]:
+
+
+    $ cd brilift
+    $ cargo build
+    $ cargo install  # If you want the executable on your $PATH.
+
+[bril-rs]: rust.md
+[cargo]: https://doc.rust-lang.org/cargo/
+
+Compile Stuff
+-------------
+
+Provide the `brilift` executable with a Bril JSON program:
+
+    $ bril2json < something.bril | brilift
```

Comment:
I tried to do this without looking at the code base(as a user) and it didn't seem to work. I tried to run `bril2json < ../benchmarks/ackermann.bril | brilift` and `bril2json < ../test/interp/tiny.bril | brilift` which both error. These programs both work when running with `brilift/run.sh` so I was surprised by this.

Reply:
The error I receive is

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unimplemented relocation addend Relocation { offset: 8, size: 26, kind: Relative, encoding: AArch64Call, symbol: SymbolId(5), addend: -4 }")', /Users/patricklafontaine/.cargo/registry/src/github.com-1ecc6299db9ec823/cranelift-object-0.83.0/src/backend.rs:500:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I'm curious why run.sh works without issue.

Reply:
Ah yes, this is the Apple Silicon trouble I alluded to in the original PR comment. I still want to work more on sorting this out, because I think it is either my fault or a simple fix on the Cranelift side—Cranelift has officially supported Apple Silicon for a while.

The reason `run.sh` works is because it specifies an x86_84 target instead of using the default architecture for the system, so everything is running through Rosetta 2.